templates/CRM/common/paymentBlock.tpl

```diff
@@ -41,7 +41,7 @@
       billing_block.show();
       cj('#billing-payment-block select.crm-select2').removeClass('crm-no-validate');
       // also set selected payment methods
-      cj('input[name="payment_processor_id"][checked=checked]').prop('checked', true);
+      cj('input[name="payment_processor_id"]:checked').prop('checked', true);
```

Comment:
I'm confused by what this code even does? It reads "Find a checkbox that's already checked, and check it". Huh?

Reply:
@I see - it was added here

https://github.com/civicrm/civicrm-core/pull/10826

the intent is to reverse

these 2 lines  - which potentially removed the property - without perhaps unchecking it?

```
   // also unset selected payment methods
      cj('input[name="payment_processor_id"]').removeProp('checked');
```

https://github.com/civicrm/civicrm-core/blob/master/templates/CRM/common/paymentBlock.tpl#L34C4-L35C70

Reply:
& in face #10826 regresses when this patch is applied

Reply:
I had a bit of a tinker, trying to set it in `.data()` instead - but I didn't nail it - however to fix this without breaking #10826 it needs to work with an amount of 0 in the mix & toggling between having a value or no value & checking it is selected OK

Reply:
Raised https://github.com/civicrm/civicrm-core/pull/34180 to fix the issue without breaking https://github.com/civicrm/civicrm-core/pull/10826 @eileenmcnaughton @monishdeb